libsolidity/formal/SMTChecker.cpp

```diff
@@ -1073,17 +1076,20 @@ smt::Expression SMTChecker::arithmeticOperation(
 		_op == Token::Add ? _left + _right :
 		_op == Token::Sub ? _left - _right :
 		_op == Token::Div ? division(_left, _right, intType) :
-		/*op == Token::Mul*/ _left * _right
+		_op == Token::Mul ? _left * _right :
+		/*_op == Token::Mod*/ _left % _right
 	);
 
-	if (_op == Token::Div)
+	OverflowTarget::Type overflowType{OverflowTarget::Type::All};
+	if (_op == Token::Div || _op == Token::Mod)
 	{
 		checkCondition(_right == 0, _location, "Division by zero", "<result>", &_right);
 		m_interface->addAssertion(_right != 0);
+		overflowType = OverflowTarget::Type::Overflow;
```

Comment:
Strictly speaking that's only the ``Div``, but not the ``Mod`` case, though, right ;-)? Shouldn't really hurt to add the check nonetheless, but maybe we shouldn't for ``Mod``...

Reply:
Ah, actually no, this is probably right because of ``-128 % -128``...

Reply:
``-128 % -128`` is rightly ``0`` without overflow... ``-128 % -1`` as well... so I don't think there's an overflow for ``Mod``.

Reply:
Actually `mod` should neither underflow nor overflow

Reply:
I think I'll just bring all checks back for all cases...

Reply:
If you don't expect them to be too expensive, then we might as well, yeah.